java/ql/src/semmle/code/java/dataflow/FlowSources.qll

```diff
@@ -318,3 +319,34 @@ class AndroidIntentInput extends DataFlow::Node {
     )
   }
 }
+
+/** Method access to external inputs of `android.content.Intent` or `android.os.BaseBundle` object. */
+class IntentGetExtraMethodAccess extends MethodAccess {
+  IntentGetExtraMethodAccess() {
+    exists(AndroidComponent ac |
+      this.getEnclosingCallable().getDeclaringType() = ac and
+      ac.isExported() and
+      this.getMethod().getName().regexpMatch("get\\w+Extra") and
+      this.getMethod().getDeclaringType() instanceof TypeIntent
+    )
+    or
+    this.getMethod().getName().regexpMatch("get\\w+") and
+    this
+        .getMethod()
+        .getDeclaringType()
+        .getASupertype*()
+        .hasQualifiedName("android.os", "BaseBundle")
+  }
+}
+
+/** Android intent extra source. */
+private class AndroidIntentExtraSource extends RemoteFlowSource {
+  AndroidIntentExtraSource() {
+    exists(AndroidIntentInput inode |
+      this.asExpr() = inode.asExpr() or
+      this.asExpr() = inode.asParameter().getAnAccess()
+    )
+  }
+
+  override string getSourceType() { result = "Android intent extra" }
+}
```

Comment:
Probably not necessary -- once `AndroidIntentInput` is reduced to only inputs of exported classes, we can simply have that class extend `RemoteFlowSource`

Reply:
Yes, it's not required once <code>AndroidIntentInput</code> is reduced to only inputs of exported classes. Changed <code>AndroidIntentInput</code> to extend <code>RemoteFlowSource</code>.